### PR TITLE
feat: add ProjectMapping config schema and TOML persistence

### LINKED ISSUE #7: feat: add ProjectMapping config schema and TOML persistence
Part of #6

## Summary

Add a `ProjectMapping` dataclass and update `MappingConfig` to support an explicit list of TickTick-to-TaskWarrior project mappings in the config file. Add a `save_config_mapping()` helper that persists mapping changes using tomlkit.

## Current Behavior

`MappingConfig` (`src/tickticksync/config.py:26-27`) has a single field:
```python
@dataclass
class MappingConfig:
    default_project: str = "inbox"
```

Config TOML:
```toml
[mapping]
default_project = "inbox"
```

## Desired Behavior

New config schema:
```toml
[mapping]
default_project = "inbox"

[[mapping.projects]]
ticktick = "Inbox"
taskwarrior = "inbox"

[[mapping.projects]]
ticktick = "Work"
taskwarrior = "work"
```

New Python types:
```python
@dataclass
class ProjectMapping:
    ticktick: str       # TickTick project name
    taskwarrior: str    # TaskWarrior project name

@dataclass
class MappingConfig:
    default_project: str = "inbox"
    projects: list[ProjectMapping] = field(default_factory=list)
```

New helper function:
```python
def save_config_mapping(path: Path, projects: list[ProjectMapping]) -> None:
    """Write or overwrite [[mapping.projects]] in config, preserving other sections."""
```

## Acceptance Criteria

- [ ] VERIFY: `ProjectMapping` dataclass exists with `ticktick` and `taskwarrior` string fields
- [ ] VERIFY: `MappingConfig.projects` is a `list[ProjectMapping]` defaulting to empty list
- [ ] VERIFY: `load_config()` correctly parses `[[mapping.projects]]` TOML array-of-tables into `MappingConfig.projects`
- [ ] VERIFY: `load_config()` works with existing configs that have no `[[mapping.projects]]` (backwards compatible)
- [ ] VERIFY: `save_config_mapping()` writes `[[mapping.projects]]` entries using tomlkit, preserving other config sections
- [ ] VERIFY: Tests cover parsing with/without mappings, round-trip write/read, and edge cases (empty list, duplicate names)

## Implementation Approach

1. Add `ProjectMapping` dataclass to `config.py`
2. Update `MappingConfig` to include `projects: list[ProjectMapping]`
3. Update `load_config()` to parse `mapping.projects` from TOML data
4. Add `save_config_mapping()` using tomlkit (follow pattern of existing `save_config_auth()`)
5. Add unit tests

### Files to Modify
- `src/tickticksync/config.py` — New dataclass, updated parsing, new save helper

## Complexity Hint
- Subsystems: Config (TOML parsing/writing)
- Structural changes: No (additive only, backwards compatible)

### In Scope
- `ProjectMapping` dataclass
- Updated `MappingConfig` with `projects` field
- `load_config()` parsing for `[[mapping.projects]]`
- `save_config_mapping()` helper
- Unit tests

### Out of Scope
- CLI commands (sub-issue #2)
- Sync engine changes (sub-issue #4)
- Validation of project names against TickTick API

### PR DESCRIPTION
**Pull request by rube-de**
## Summary

Closes #7 (part of #6)

- Add `ProjectMapping` dataclass with `ticktick` and `taskwarrior` string fields
- Update `MappingConfig` with `projects: list[ProjectMapping]` (defaults to empty list)
- Update `load_config()` to parse `[[mapping.projects]]` TOML array-of-tables into typed objects
- Add `save_config_mapping()` helper using tomlkit for format-preserving writes
- Backwards compatible: existing configs without `[[mapping.projects]]` continue to work

## Test plan

- [x] 10 new tests added (21 total config tests, 88 full suite)
- [x] Roundtrip write/read verified
- [x] Preserves other config sections (ticktick, sync, auth)
- [x] Empty list clears mappings
- [x] Overwrite existing mappings
- [x] Duplicate project names allowed (no validation at config layer)
- [x] Create config from scratch (new file)
- [x] Zero regressions on full test suite

### CODE REVIEW

File: src/tickticksync/config.py [hunk 1]
```diff
@@ -88,3 +98,22 @@ def save_config_auth(path: Path, method: AuthMethod, username: str | None = None
...
+    if "mapping" not in doc:
+        doc.add("mapping", tomlkit.table())
+    aot = tomlkit.aot()
+    for pm in projects:
+        t = tomlkit.table()
+        t.add("ticktick", pm.ticktick)
+        t.add("taskwarrior", pm.taskwarrior)
+        aot.append(t)
+    doc["mapping"]["projects"] = aot
+    path.write_text(tomlkit.dumps(doc))
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/tickticksync/config.py]: Fixed: added path.parent.mkdir(parents=True, exist_ok=True) before write_text()

File: src/tickticksync/config.py [hunk 2]
```diff
@@ -64,10 +71,13 @@ def load_config(path: Path | None = None) -> Config:
         raise ValueError(
             f"Invalid auth.method {method!r} in config; expected one of {_VALID_AUTH_METHODS}"
         )
+    mapping_data = data.get("mapping", {})
+    projects_raw = mapping_data.pop("projects", [])
+    projects = [ProjectMapping(**p) for p in projects_raw]
```
[the remark this answers is not in the document]
Comment (rube-de) [reply src/tickticksync/config.py]: Fixed: replaced list comprehension with enumerate loop that catches TypeError and re-raises as ValueError with index and expected keys
